File: 1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.py
class Solution:
    def average(self, salary: List[int]) -> float:
        # O(1) time & O(1) space
        m = min(salary)
        M = max(salary)
        s = sum(salary)
        n = len(salary)
        return (s - M - m) / (n - 2)

File: 1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.java
class Solution {
    public double average(int[] salary) {
        // O(n) time & O(1) space
        
        Arrays.sort(salary);
        
        double sum = 0;
        int n = salary.length - 2;
        for(int i = 1; i <= n; i++) sum+= salary[i];
        
        double avg = sum / n;
        
        return avg;
    }
}